routes/intallatievergadering.ts

```diff
@@ -0,0 +1,229 @@
+import { Request, Response } from 'express';
+import Router from 'express-promise-router';
+import { query, sparqlEscapeString, sparqlEscapeUri } from 'mu';
+import { updateSudo, querySudo } from '@lblod/mu-auth-sudo';
+import { v4 as uuidv4 } from 'uuid';
+
+const installatievergaderingRouter = Router();
+
+installatievergaderingRouter.post(
+  '/:id/move-ocmw-organs/',
+  async (req: Request, res: Response) => {
+    const installatievergaderingId = req.params.id;
+    const hasAccess = await canSeeInstallatievergadering(
+      installatievergaderingId,
+    );
+    if (!hasAccess) {
+      return res
+        .status(404)
+        .send({ error: 'Installatievergadering not found' });
+    }
+    await moveFracties(installatievergaderingId);
+    await moveOcmwOrgans(installatievergaderingId);
+    return res.status(200).send({ status: 'ok' });
+  },
+);
+
+async function canSeeInstallatievergadering(id: string) {
+  const sparql = `
+  PREFIX ext: <http://mu.semte.ch/vocabularies/ext/>
+  PREFIX mu: <http://mu.semte.ch/vocabularies/core/>
+
+  SELECT * WHERE {
+    ?s a ext:Installatievergadering .
+    ?s mu:uuid ${sparqlEscapeString(id)} .
+  } LIMIT 1`;
+  const result = await query(sparql);
+  return result.results.bindings.length > 0;
+}
+
+async function moveFracties(installatievergaderingId: string) {
+  const hasExistingFractions = await ocmwHasFractions(installatievergaderingId);
+  if (hasExistingFractions) {
+    console.log(
+      `Cowardly refusing to create fractions for OCMW of ${installatievergaderingId} as it already has fracties`,
+    );
+    return;
+  }
+  const existingFractionsGemeente = await getExistingGemeenteFractions(
+    installatievergaderingId,
+  );
+
+  const newFractions = existingFractionsGemeente.map((fraction) => {
```

Comment:
Should this actually be new fracties with new uuids and uris, not really sure what the business rules about this are.

Reply:
i checked with niels and yes they should be :(